scrabble.js

```diff
@@ -1,16 +1,137 @@
-
 const Scrabble = {
+  //O(1)
   score(word) {
+    let scoreChart = {
+      'A': 1,
+      'E': 1,
+      'I': 1,
+      'O': 1,
+      'U': 1,
+      'L': 1,
+      'N': 1,
+      'R': 1,
+      'S': 1,
+      'T': 1,
+      'D': 2,
+      'G': 2,
+      'B': 3,
+      'C': 3,
+      'M': 3,
+      'P': 3,
+      'F': 4,
+      'H': 4,
+      'V': 4,
+      'W': 4,
+      'Y': 4,
+      'K': 5,
+      'J': 8,
+      'X': 8,
+      'Q': 10,
+      'Z': 10
+    };
+    let wordCount = 0;
+    if ( word.length === 0 || word.length > 7 ) {
+      throw 'Invalid word';
+    }
+
+    let upperCase = word.toUpperCase().split('')
+    if ( upperCase.length === 7 ) {
+      wordCount += 50;
+    }
 
+    upperCase.forEach(function(letter) {
+      if ( scoreChart[letter] === undefined ) {
+        throw 'Invalid character(s)!';
+      } else {
+        wordCount += scoreChart[letter];
+      }
+    })
+    return wordCount;
   },
+  //O(mn) where m is the length of the array and n is the length of the word
   highestScoreFrom(arrayOfWords) {
+    let scores = {};
+    let nums = [];
+
+    if ( arrayOfWords.length === 0 ) {
+      throw 'No words!';
+    }
+
+    arrayOfWords.forEach( function(word) {
+      let score = Scrabble.score(word);
 
+      if ( scores[score] === undefined ) { //new score
+        scores[score] = word;
+        nums.push(score);
+      } else { //tied score
+        if ( scores[score].length != 7 ) {
+          if ( scores[score].length >  word.length ) {
+            scores[score] = word;
+          } else if ( scores[score].length <  word.length && word.length === 7 ) {
+            scores[score] = word;
+          }
+        }
+      }
+    });
+
+    let max_score = Math.max(...nums);
+    return scores[max_score];
```

Comment:
Interesting + good solution to use a hash and an array!

I haven't thought through it too deeply, but I wonder if there's a way to keep this solution while avoiding making the array `nums`... Your `scores` object is already keeping track of scores and words, so maybe you don't need it!
You really only use `nums` to find the `max_score`, so maybe there's a way to get the numerical scores out of `scores` and sort those and find the max...